File: booking-service/src/main/java/entelect/training/incubator/spring/booking/api/FlightApiClient.java
package entelect.training.incubator.spring.booking.api;

import org.springframework.stereotype.Component;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.reactive.function.client.WebClient;
import reactor.core.publisher.Mono;

@Component
public class FlightApiClient {

    private final WebClient webClient = WebClient.create("http://localhost:8202");


    @GetMapping("/{id}")
    public void verifyFlightExists(String flightId) {
         webClient.get()
                .uri("/flights/{id}", flightId)
                .retrieve()
                .toBodilessEntity().block();
    }

}

File: booking-service/src/main/kotlin/entelect/training/incubator/spring/booking/api/FlightApiClient.kt
package entelect.training.incubator.spring.booking.api

import org.springframework.http.HttpHeaders
import org.springframework.stereotype.Component
import org.springframework.web.reactive.function.client.WebClient
import reactor.core.publisher.Mono

@Component
class FlightApiClient {

    private val webClient: WebClient = WebClient.builder().baseUrl("http://localhost:8202")
        .defaultHeaders(
        {
            headers: HttpHeaders? ->
            headers!!.setBasicAuth("user", "the_cake")
        }
    ).build()

    fun verifyFlightExists(flightId: String): Mono<Void>{
        return webClient.get().uri("/flights/{id}", flightId)
            .retrieve().toBodilessEntity().then();
    }
}